### PR TITLE
Let Wingman peek through type families

### LINKED ISSUE #1874: Wingman can't split type family
```haskell
{-# LANGUAGE TypeFamilies #-}

module Lib where

type family Yo where
    Yo = Bool

test :: Yo -> Int
test b = _
```

Running `Destruct b` at the hole prints an internal error, I guess because there are no data constructors. Pretty sure this used to work, so not sure when it regressed.

### PR DESCRIPTION
**Pull request by isovector**
This PR fixes a bug where type families would be opaque to Wingman, even if they weren't stuck. Now, we normalize any types coming into the judgment.

Fixes #1874

### CODE REVIEW

File: plugins/hls-tactics-plugin/test/golden/DestructTyFam.hs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{-# LANGUAGE TypeFamilies #-}
+
+type family Yo where
+  Yo = Bool
+
+test :: Yo -> Int
+test b = _
+
```
Comment (Ailrun) [inline plugins/hls-tactics-plugin/test/golden/DestructTyFam.hs:8]: Does this work with open type family too?

Comment (isovector) [reply plugins/hls-tactics-plugin/test/golden/DestructTyFam.hs:8]: Added some tests

Comment (Ailrun) [reply plugins/hls-tactics-plugin/test/golden/DestructTyFam.hs:8]: Awesome, thanks!

**Review by Ailrun — APPROVED** [review]
LGTM :)

**Pull request merged**